photon-client/src/main/scala/com/linkedin/photon/ml/data/DataValidators.scala

```diff
@@ -184,10 +187,11 @@ object DataValidators extends Logging {
    * @param inputColumnName The column name we want to validate
    * @return Whether the weight of the input data point is finite
    */
-  def rowHasFiniteWeight(row: Row, inputColumnName: String): Boolean = {
-    val weight = row.getAs[Double](inputColumnName)
-    !(weight.isNaN || weight.isInfinite)
-  }
+  def rowHasFiniteWeight(row: Row, inputColumnName: String): Boolean =
+    row.getAs[Any](inputColumnName) match {
+      case weight: Double => !(weight.isNaN || weight.isInfinite)
+      case _ => false
+    }
```

Comment:
The functions `rowHasFiniteLabel`, `rowHasFiniteOffset`, and `rowHasFiniteWeight` are really the same function. You can reduce it down to one function called `rowHasFiniteValue` or something like that.